server/src/QRServer/common/classes.py

```diff
@@ -92,10 +92,16 @@ def add_party(self, party: MatchParty):
             party.match_opponent(self.parties[0])
             self.parties[0].match_opponent(party)
 
+    def has_party(self, party: MatchParty):
+        return party in self.parties
+
     def remove_party(self, party: MatchParty):
         if party not in self.parties:
             raise Exception('Cannot find the given party')
 
+        if not len(self.match_stats) == 2:
```

Comment:
Hacky way to retain user_id after user disconnects, since its not available anywhere else